Quick_Deploy/HuggingFaceTransformers/README.md

```diff
@@ -0,0 +1,150 @@
+
+The following tutorial demonstrates how to deploy an arbitrary hugging face transformer
+model on the Triton Inference Server using Triton's [Python backend](https://github.com/triton-inference-server/python_backend). For the purposes of this example, three transformer
+models will be deployed:
+- [facebook/opt-125m](https://huggingface.co/facebook/opt-125m) for text generation requests.
+- [gpt2](https://huggingface.co/gpt2) for text generation requests.
+- [distilbert-base-uncased-finetuned-sst-2-english](https://huggingface.co/distilbert-base-uncased-finetuned-sst-2-english) for text classification requests.
+
+These models were selected because of their small size and consistent response quality, 
+enabling the tutorial to run on most devices with sensible results. However, this tutorial
+is also generalizable for much larger models provided sufficient infrastructure. 
+
+*NOTE*: The tutorial is intended to be a reference example only. It is a work in progress.
+
+## Step 1: Build a Model Repository
+
+The first step is to create a model repository that the Triton Inference Server will use
+for inference processing. We can create a model repository from the provided base files by 
+executing the following python script:
+
+```
+python3 create_repository.py
+```
+
+Without specifying any arguments, this script will create a model repository containing
+the details necessary for Triton to load and serve each of the aforementioned models.
+For the purposes of this tutorial, we provide Triton with simple ```config.pbtxt``` files
+for each model that describe:
+- The the backend to use.
+- Model input and output details.
+- Custom parameters to use for execution.
+
+## Step 2: Build a Triton Container Image
+
+The second step is to create an image that includes all the dependencies necessary
+to deploy hugging face transformer models on the Triton Inference Server. This can be done
+by building an image from the provided Dockerfile.
+
+```
+docker build -t triton_transformer_server .
+```
+
+## Step 3: Run the Triton Inference Server
+
+Once the ```triton_transformer_server``` image is created, you can run a container with
+the following command:
+
+```bash
+docker run --gpus all -it --rm -p 8000:8000 --shm-size=1G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}/model_repository:/opt/tritonserver/model_repository triton_transformer_server tritonserver --model-repository=model_repository
+```
+
+The server has launched successfully when you see the following outputs in your console:
+
+```bash
+I0922 23:28:40.351809 1 grpc_server.cc:2451] Started GRPCInferenceService at 0.0.0.0:8001
+I0922 23:28:40.352017 1 http_server.cc:3558] Started HTTPService at 0.0.0.0:8000
+I0922 23:28:40.395611 1 http_server.cc:187] Started Metrics Service at 0.0.0.0:8002
+```
+
+## Step 4: Use a Triton Client to Query the Server
+
+Now we can query the server using the client script provided.
+
+```bash
+python3 client.py -m [model_name] --prompt [prompt_message]
```

Comment:
can we add curl as reference?

Reply:
This would be blocked until Guan and I finish implementing simpler curl format support, unless you mean a tensor-oriented curl:
```
curl -X POST localhost:8000/v2/models/text/infer -d '{"inputs": [{"name":"prompt","datatype":"BYTES","shape":[1,1],"data":["hello"]}]}'
```

Reply:
I think the simple curl one liner is good for now - and we will add the generate version when available